mathics/core/list.py

```diff
@@ -98,16 +102,17 @@ def evaluate_elements(self, evaluation):
                         elements_changed = True
                         elements[index] = new_value
 
-        if elements_changed:
-            # Save changed elements, making them immutable again.
-            self._elements = tuple(elements)
+        if not elements_changed:
+            return self
 
-            # TODO: we could have a specialized version of this
-            # that keeps self.value up to date when that is
-            # easy to do. That is left of some future time to
-            # decide whether doing this this is warranted.
-            self._build_elements_properties()
-            self.value = None
+        new_list = ListExpression(*elements)
+        # TODO: we could have a specialized version of this
+        # that keeps self.value up to date when that is
+        # easy to do. That is left of some future time to
+        # decide whether doing this this is warranted.
+        new_list._build_elements_properties()
+        new_list.value = None
+        return new_list
```

Comment:
Notice that in `Expression.evaluate_elements`, a new expression is returned.

Reply:
Thanks for catching. I guess this was a bug fixed then. If this was causing a problem, I am curious as to where it was causing a problem.